### PR TITLE
[linmath] add new port 

### LINKED ISSUE #24657: Add the linmath.h to the list of packages available
****

**Is your feature request related to a problem? Please describe.**
I want to install linmath.h for use with glfw opengl library and it is not available. I think it should be added.

**Proposed solution**
I want linmath.h to added to the list of available packages.

**Describe alternatives you've considered**
I have not explored alternatives.

**Additional context**
not neccessary.

### PR DESCRIPTION
**Pull request by RyanPaulMcKenna**
**Describe the pull request**
 add port for linmath.h
- #### What does your PR fix?
  Fixes #...
It make the header only library linmath.h available via vcpkg
- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  <all / linux, windows, ...>, <Yes/No>
No
- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
 Yes I believe so.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
It is a new port with only one version so no update is required.

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**
It is complete.

Fixes https://github.com/microsoft/vcpkg/issues/24657

### CODE REVIEW

File: ports/linmath/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,13 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO datenwolf/linmath.h
+    REF refs/heads/master
```
Comment (Thomas1664) [inline ports/linmath/portfile.cmake:4]: This would invalidate the hash on upstream updates. Please use the last commit id instead.

**Review by Thomas1664 — CHANGES_REQUESTED** [review]

**Comment by RyanPaulMcKenna** [comment]
This pull request solves the following:
https://github.com/microsoft/vcpkg/issues/24657

File: ports/linmath/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/linmath/portfile.cmake:4]: @RyanPaulMcKenna ,Thanks for your pr,I agree with @Thomas1664 , we should use a commit id instead

File: ports/linmath/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,13 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO datenwolf/linmath.h
+    REF refs/heads/master/0538757
```
Comment (Thomas1664) [inline ports/linmath/portfile.cmake:4]:
```suggestion
    REF 3954f97604a522c4b9c0268188f1a3d47ad2a205
```
This would be the correct way

Comment (RyanPaulMcKenna) [reply ports/linmath/portfile.cmake:4]: Oh okay. Where did you get that?

Comment (RyanPaulMcKenna) [reply ports/linmath/portfile.cmake:4]: Okay I checked It is the permalink.

Comment (Thomas1664) [reply ports/linmath/portfile.cmake:4]: For reference: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_from_github.md

Comment (dg0yt) [reply ports/linmath/portfile.cmake:4]: Branches are expected to move over time. So the SHA checksum would no longer match. Tags are expected to be stable names.

**Comment by RyanPaulMcKenna** [comment]
Should this PR be closed?

File: ports/linmath/portfile.cmake (hunk 2 above)
Comment (JackBoosY) [reply ports/linmath/portfile.cmake:4]: I think it's good.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
> Should this PR be closed?

@RyanPaulMcKenna, No, Please revise according to this review

> Should this PR be closed?

No, I have approved this pr, Thanks!

File: ports/linmath/portfile.cmake (hunk 2 above)
Comment (ras0219-msft) [reply ports/linmath/portfile.cmake:4]: This should be replaced with the stable reference.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
LGTM after switching to commit sha. Thanks for the PR!

**Comment by JonLiu1993** [comment]
@RyanPaulMcKenna, Ping for response

**Comment by unjello** [comment]
Any chances this may get any traction?

**Comment by Thomas1664** [comment]
> Any chances this may get any traction?

Since the author didn't respond to the requested changes and people actually want this port I finished this PR in #26069.

**Comment by JonLiu1993** [comment]
Duplicate of #26069

**Pull request closed without merging**